src/popup/popup.js

```diff
@@ -1 +1,60 @@
-/* popup.js */
+import {
+  getAllExtensions,
+  areExtsBeingMonitored,
+  initMonitorAll,
+  stopMonitorAll,
+  viewActivityLogs,
+} from '../lib/ext-listen.js';
+
+async function init() {
+  const extensions = await getAllExtensions();
+  const startMonitorAllBtn = document.getElementById('startMonitorBtn');
+  const stopMonitorAllBtn = document.getElementById('stopMonitorBtn');
+  const viewActivityLogBtn = document.getElementById('actLogPage');
+  const monitorStatusText = document.getElementById('monitorStatus');
+
+  startMonitorAllBtn.addEventListener('click', async () => {
+    const monitorMsg = await initMonitorAll(extensions);
```

Comment:
In this version of this PR, the popup seems to be retrieving the list of all the extensions installed when it is initialized, but the only purpose of doing it seems to be to send the array of the extensions to the background page.
The background page could just call getAllExtensions internally, and the popup could just send a message to ask the background page to monitor all installed extensions.

Reply:
Done :-) Extensions are being fetched in the background now.